features/neuropixel_features.py

```diff
@@ -0,0 +1,141 @@
+class OpenEphysHTTPServer:
+    def __init__(self, address='10.155.205.108',timeout=0.5):
+        self.address = 'http://' + address + ':37497'
+        self.timeout = timeout
+
+    def send(self, endpoint, payload=None):
+        # Send a request to the server
+        try:
+            if payload is None:
+                resp = requests.get(self.address + endpoint, timeout=self.timeout)
+            else:
+                resp = requests.put(self.address + endpoint, data = json.dumps(payload), timeout=self.timeout)
+            response = resp.json()
+        except Exception as e:
+            print(e, 'Check if you started openephys in neuropixel computer')
+            resp = {}
+            resp['mode'] = 'No connection with openephys'
+            response = resp
+        return response
+
+    def set_parent_dir(self,path):
+        # Set the parent directory
+        payload = {'parent_directory' : path}
+        data = self.send('/api/recording', payload=payload)
+        return data
+
+    def set_prepend_text(self,text):
+        # Set the prepend name for the recording directory
+        payload = {'prepend_text' : text}
+        data = self.send('/api/recording', payload=payload)
+        return data
+
+    def set_base_text(self,text):
+        # Set the base name for the recording directory
+        payload = {'base_text' : text}
+        data = self.send('/api/recording', payload=payload)
+        return data
+
+    def set_append_text(self,text):
+        # Set the append name for the recording directory
+        payload = {'append_text' : text}
+        data = self.send('/api/recording', payload=payload)
+        return data
+
+    def status(self):
+        # Get the current status of the GUI (IDLE, ACQUIRE, or RECORD)
+        return self.send('/api/status')['mode']
+
+    def acquire(self):
+        # Start acquisition
+        payload = { 'mode' : 'ACQUIRE',}
+        data = self.send('/api/status', payload=payload)
+        return data['mode']
+
+    def record(self):
+        # Start recording data
+        previous_mode = self.status()
+        payload = {'mode' : 'RECORD',}
+        data = self.send('/api/status', payload=payload)       
+        return data['mode']
+
+    def idle(self):
+        # Stop acquisition or recording data
+        previous_mode = self.status()
+        payload = { 'mode' : 'IDLE',}
+        data = self.send('/api/status', payload=payload)
+        return data['mode']
+
+
+class RecordNeuropixels(traits.HasTraits):
+
+    neuropixel_port1 = traits.Int(0, desc="Channel number you used for recoding with port 1 probe")
+    neuropxiel_port2 = traits.Int(0, desc="Channel number hole you used for recoding with port 2 probe")
+    neuropixel_mapping_file = traits.String("", desc="Name of channel mapping file")
+
+    IP_neuropixel = '10.155.205.108'
+    gui = OpenEphysHTTPServer(IP_neuropixel, timeout=0.5)
+    parent_dir = 'E://Neuropixel_data'
+    gui.set_parent_dir(parent_dir)
```

Comment:
Can you move these lines into pre_init() and cleanup() instead of here? They will run every time someone opens BMI3D, even if they aren't using neuropixels, because this code will run any time features is imported.

Reply:
Instantiating `OpenEphysHTTPServer` itself doesn't make a connection with openephys. But `gui.set_parent _dir` tries to connect with openephys. So, I only moved `gui.set_parent_dir` to `pre_init.`

Reply:
Makes sense